packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/tile_overlay.dart

```diff
@@ -80,14 +80,14 @@ class TileOverlay {
   final bool fadeIn;
 
   /// The tile provider to use for this tile overlay.
-  final TileProvider tileProvider;
+  final TileProvider? tileProvider;
 
   /// The transparency of the tile overlay. The default transparency is 0 (opaque).
   final double transparency;
 
   /// The tile overlay's zIndex, i.e., the order in which it will be drawn where
   /// overlays with larger values are drawn above those with lower values
-  final int zIndex;
+  final int? zIndex;
```

Comment:
@cyanglaz Is there a reason there's no default value for zIndex? Seems odd to have it allow null rather than have a default (e.g., 0)